src/core/slices/mainpage.ts

```diff
@@ -8,6 +8,7 @@ type State = {
   };
   queueStatus: QueueStatusType;
   banStatus: AniDBBanType;
+  queueModalOpen: boolean;
```

Comment:
Because then we would have two modals mounted to the react tree on the dashboard, and one modal everywhere else. The modal is only mounted in the top-nav component, but the dispatch to open it is in two components, the top-nav and the dashboard queue processor widget.

Also the icon in the top-nav will light up when you open the modal from the dashboard widget if we store the state in a shared place. We won't be able to do that if we only store the state locally in the parent component of the modal.

Reply:
Fair